src/containers/Cinema/Cinema.tsx

```diff
@@ -31,25 +28,27 @@ type Props = {
 };
 
 const Cinema: React.FC<Props> = ({ item, onPlay, onPause, onComplete, onUserActive, onUserInActive, feedId, isTrailer = false }: Props) => {
-  const config: Config = useContext(ConfigContext);
+  const { player, continue_watching_list } = useConfigStore(({ config }) => ({
+    player: config.player,
+    continue_watching_list: config.features?.continue_watching_list,
+  }));
   const playerElementRef = useRef<HTMLDivElement>(null);
   const playerRef = useRef<JWPlayer>();
   const loadingRef = useRef(false);
   const seekToRef = useRef(-1);
   const [libLoaded, setLibLoaded] = useState(!!window.jwplayer);
-  const scriptUrl = `https://content.jwplatform.com/libraries/${config.player}.js`;
-  const enableWatchHistory = config.content.some((el) => el.type === PersonalShelf.ContinueWatching) && !isTrailer;
+  const scriptUrl = `https://content.jwplatform.com/libraries/${player}.js`;
+  const enableWatchHistory = continue_watching_list && !isTrailer;
   const setPlayer = useOttAnalytics(item, feedId);
   const handlePlaylistItemCallback = usePlaylistItemCallback();
 
-  const getProgress = useCallback((): VideoProgress | null => {
+  const getProgress = useCallback((): number | null => {
     if (!playerRef.current) return null;
 
-    const duration = playerRef.current.getDuration();
-    const progress = playerRef.current.getPosition() / duration;
+    const progress = playerRef.current.getPosition() / item.duration;
```

Comment:
Is there any chance that item.duration will be wrong, less precise, or missing?

Reply:
I didn't notice any issues if we don't store the duration in externalData but just retrieve it from the media object.

Reply:
OK, then nice optimization!